hmac.h

```diff
@@ -36,7 +36,19 @@ typedef sha256_context_t dtls_sha256_ctx;
 #define dtls_sha256_update(Ctx,Input,Len) sha256_update((Ctx), (Input), (Len))
 #define dtls_sha256_final(Buf,Ctx)        sha256_final((Ctx), (Buf))
 
-#else /* RIOT_VERSION */
+#elif defined ESPIDF_VERSION
```

Comment:
The pre-processor symbol `ESPIDF_VERSION` is set in a different component (libcoap) and thus not very generic. 
I suggest testing for `ESP_PLATFORM` instead. Moreover, I presume the presence of libsodium should be checked as well.

Reply:
Fixed